src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -2545,8 +2914,30 @@ public ResourceEvent(IResource resource, string resourceId, CustomResourceSnapsh
         public CustomResourceSnapshot Snapshot { get { throw null; } }
     }
 
+    public static partial class ResourceExecutionConfigurationBuilderExtensions
+    {
+        public static IResourceExecutionConfigurationBuilder WithArgumentsConfig(this IResourceExecutionConfigurationBuilder builder) { throw null; }
+
+        [System.Diagnostics.CodeAnalysis.Experimental("ASPIRECERTIFICATES001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+        public static IResourceExecutionConfigurationBuilder WithCertificateTrustConfig(this IResourceExecutionConfigurationBuilder builder, System.Func<CertificateTrustScope, CertificateTrustExecutionConfigurationContext> configContextFactory) { throw null; }
+
+        public static IResourceExecutionConfigurationBuilder WithEnvironmentVariablesConfig(this IResourceExecutionConfigurationBuilder builder) { throw null; }
+
+        [System.Diagnostics.CodeAnalysis.Experimental("ASPIRECERTIFICATES001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+        public static IResourceExecutionConfigurationBuilder WithHttpsCertificateConfig(this IResourceExecutionConfigurationBuilder builder, System.Func<System.Security.Cryptography.X509Certificates.X509Certificate2, HttpsCertificateExecutionConfigurationContext> configContextFactory) { throw null; }
+    }
+
+    public static partial class ResourceExecutionConfigurationExtensions
+    {
+        public static bool TryGetAdditionalData<T>(this IProcessedResourceExecutionConfiguration configuration, out T? additionalData)
+            where T : IResourceExecutionConfigurationData { throw null; }
+    }
+
     public static partial class ResourceExtensions
     {
+        public static IResourceExecutionConfigurationBuilder ExecutionConfigurationBuilder(this IResource resource) { throw null; }
```

Comment:
@danegsta - This method name isn't a verb. Should it be `CreateExecutionConfigurationBuilder`?

Do we need this as an extension method? Or can it be a static method?

Reply:
Let's remove the extension method and make ResourceExecutionConfigurationBuilder.Create public (which would be renamed to just ExecutionConfigurationBuilder.Create with https://github.com/dotnet/aspire/pull/13290#discussion_r2612163996